codalab/lib/upload_manager.py

```diff
@@ -36,57 +36,52 @@ def upload_to_bundle_store(
+        - If |git|, then the bundle contains the result of running 'git clone |source|'
         - If |unpack| is True or a source is an archive (zip, tar.gz, etc.), then unpack the source.
         """
         bundle_path = self._bundle_store.get_bundle_location(bundle.uuid)
         try:
             path_util.make_directory(bundle_path)
-            # Note that for uploads with a single source, the directory
-            # structure is simplified at the end.
-            for source in sources:
-                is_url, is_fileobj, filename = self._interpret_source(source)
-                source_output_path = os.path.join(bundle_path, filename)
-                if is_url:
-                    assert isinstance(source, str)
-                    if git:
-                        source_output_path = file_util.strip_git_ext(source_output_path)
-                        file_util.git_clone(source, source_output_path)
-                    else:
-                        file_util.download_url(source, source_output_path)
-                        if unpack and self._can_unpack_file(source_output_path):
-                            self._unpack_file(
-                                source_output_path,
-                                self.zip_util.strip_archive_ext(source_output_path),
-                                remove_source=True,
-                                simplify_archive=simplify_archives,
-                            )
-                elif is_fileobj:
-                    if unpack and self.zip_util.path_is_archive(filename):
-                        self._unpack_fileobj(
-                            source[0],
-                            source[1],
+            # Note that the directory structure is simplified at the end.
+            is_url, is_fileobj, filename = self._interpret_source(source)
+            source_output_path = os.path.join(bundle_path, filename)
+            if is_url:
+                assert isinstance(source, str)
+                if git:
+                    source_output_path = file_util.strip_git_ext(source_output_path)
+                    file_util.git_clone(source, source_output_path)
+                else:
+                    file_util.download_url(source, source_output_path)
+                    if unpack and self._can_unpack_file(source_output_path):
+                        self._unpack_file(
+                            source_output_path,
                             self.zip_util.strip_archive_ext(source_output_path),
+                            remove_source=True,
                             simplify_archive=simplify_archives,
                         )
-                    else:
-                        with open(source_output_path, 'wb') as out:
-                            shutil.copyfileobj(cast(IO, source[1]), out)
-
-            if len(sources) == 1:
-                self._simplify_directory(bundle_path)
+            elif is_fileobj:
```

Comment:
if we remove support for multiple URLs, wouldn't we be effectively only allowing one fileobj to be uploaded? since the `for source in sources` doesn't exist - where is the multiple filenames logic being handled?

Reply:
oh, is it tarred into a single file?

Reply:
Yeah, the bundle cli tars it into a single file.